### ISSUE TITLE
removing test and re-registering it is not possible

### ISSUE THREAD

**Issue by gernotstarke** [corona-warn-app/cwa-app-ios#1198]
## Avoid duplicates

* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is not already reported in another issue

## Describe the bug

1. I registered a test via QR-code, successfully.
2. for 36hrs, I received the message "Ergebnis liegt noch nicht vor" when pressing the "aktualisieren" button in the app
3. I became impatient, and removed the test (hoping that re-registering it would finally give me a result)
4. I scanned the QR-code again, on the same mobile device.
5. I received "Fehler: Der QR-Code ist ungültig oder wurde bereits auf einem Smartphone registriert".

## Expected behaviour

in step 5 above I can register the test again, as I'm the rightful owner of the QR-code

## Steps to reproduce the issue

see above

## Technical details

- iOS Version: 14
- Device: iPhone 11 Pro

## Possible Fix

most likely a server issue?

## Additional context

Yes I know, people shouldn't be impatient when waiting for test results, but the tester (at CGN-airport) told me that results should be available after 24hrs...

**Comment by Ein-Tim** [comment]
This Issue was already rased before. 
It is not possible to have a test result registered 2 times.
@daimpi do you have the statement from one of the devs, I couldn't find it 😅

**Comment by gernotstarke** [comment]
Thanx @Ein-Tim , but that is imho not user-friendly. I read the appropriate sentence in the FAQ, but that does not solve the problem... There will be NO letter from "Gesundheitsamt" in case of negative test results. 

Speed of results-delivery should be a high-priority goal... and printed letters are not known for their "runtime performance".

And, as discussed in the thread (@thomasaugsten ), usability should be improved to at least warn (impatient) users like myself that they are about to invalidate their QR-code. 

Regarding to your question: No, I did NOT get any warning dialog/message, test was immediately removed when I hit the button. Cannot create screenshot, though...

**Comment by gernotstarke** [comment]
I really appreciate your immediate response! Shall I close the issue, or will you close it?

**Comment by Ein-Tim** [comment]
Please leave it open for a bit (just btw I'm not a dev so I can't close it)...
@thomasaugsten Why didn't @gernotstarke see a Message during removing the test? Can you confirm the Message is still there in 1.3.0?

@gernotstarke Which CWA Version are you using?

**Comment by gernotstarke** [comment]
I'm using 1.3.0 (6)

**Comment by Ein-Tim** [comment]
Okay, so if the warning wasn't removed (which would not make sense) than you should saw it... Please clarify @thomasaugsten

**Comment by gernotstarke** [comment]
obviously I cannot double-check it with my own code. In case you can provide a test-qr-code (here or by DM), I will try to reproduce.

**Comment by thomasaugsten** [comment]
The message is still in the app.
![RPReplay_Final1600564827 (2)](https://user-images.githubusercontent.com/6367738/93692261-24d19a00-faf1-11ea-940b-65efba2a45c3.gif)

You can check this by your own when you generate a qr code with the content:
 test?<randomnumber>

**Comment by gernotstarke** [comment]
thanx. I must have been **very** impatient to overlook that message.

**Comment by daimpi** [comment]
@thomasaugsten the QR codes with `test?randomnumber` don't seem to work anymore in CWA 1.5 is this correct? Is there another way to generate QR codes for test purposes?

**Comment by roussec** [comment]
I get the same error message however I did not remove the code manually but updated the Corona App which removed the code.

**Comment by Ein-Tim** [comment]
@roussec 

You mean the update to version 2.0 removed a registered test result? 
Was the test result pending or was your result already delivered (negative, positive)?

**Comment by roussec** [comment]
Yes. This is what happend. The result was/is pending. I will have to call the doctor on monday and ask if he is able to get Information about the result.

**Comment by Ein-Tim** [comment]
@roussec 

Could you please [open a new bug report](https://github.com/corona-warn-app/cwa-app-ios/issues/new?assignees=&labels=bug&template=01_bugs.md) for this?
This should be investigated by the developers.

**Comment by Ein-Tim** [comment]
@dsarkar FYI

**Comment by niz11** [comment]
This is the worse design I ever encountered in my life. 

Open the Bug again!

**Comment by Ein-Tim** [comment]
@roussec 

It's not a bug, it's a feature. In this wishlist issue I explained why this choice was met: https://github.com/corona-warn-app/cwa-wishlist/issues/389

**Comment by niz11** [comment]
It's just bad, call it as you want to call it. 

There MUST be a way to register your QR code again for a second time.

**Comment by Ein-Tim** [comment]
@roussec 

Please open a new issue for this [here](https://github.com/corona-warn-app/cwa-wishlist/issues/new/choose).

**Comment by jucktnich** [comment]
@Ein-Tim I can't find something under https://github.com/corona-warn-app/cwa-wishlist/issues/389

**Comment by Ein-Tim** [comment]
@jucktnich 

Thx for the hint, was the wrong link: https://github.com/corona-warn-app/cwa-wishlist/issues/426

**Comment by niz11** [comment]
I opened a feature request, but I still believe the current issue here should be reopened.

**Comment by Ein-Tim** [comment]
@heinezen @dsarkar FYI

**Comment by dsarkar** [comment]
@niz11 @Ein-Tim @jucktnich Thanks for the contributions here. 

That the QR code can be scanned only once has technical reasons. This specific QR code is used to transfer the test result and then eventually to share a positive result with other CWA users, if the user agrees to share. If this QR code could be scanned several times, one could manipulate the system with duplicate positives, which is obviously not desired.

Currently you have the rolling workaround, at least if you requested at the same time an EU DCC: Save the EU DCC, make a screenshot, print it. This EU DCC can be scanned again and again.

Hopefully we see in near future that such EU DCC for tests are issued directly from the test centres to the user. 

---
Corona-Warn-App Open Source Team

**Comment by posalex** [comment]
@Ein-Tim it is not a feature, it is a bug actually. I have to use luca app now, because I cannot get my tests into the coronawarn app anymore.
It seems, after I once removed the first test, I cannot add any tests from the same test lab anymore.
I have done two test already, the first I have removed, but the app does not want to import the second test, too.
It seems, I cannot add any tests from the "Testzentrum Kurt-Schumacher Straße" anymore.
Please fix this.

**Comment by posalex** [comment]
@Ein-Tim it is not a feature, it is a bug actually. I have to use luca app now, because I cannot get my tests into the coronawarn app anymore.
It seems, after I once removed the first test, I cannot add any tests from the same test lab anymore.
I have done two test at this particular testing facility already, the first one I could import but I removed it afterwards, now the app does not want to import the second test.
It seems, I cannot add any tests from the "Testzentrum Kurt-Schumacher Straße" anymore.
Maybe they just update the first certificate and do not generate a new one for each test, I do not know.
Please fix this.

**Comment by Ein-Tim** [comment]
@posalex 

Do you receive an error message?

**Comment by posalex** [comment]
@Ein-Tim 
![BFCC3485-8ECF-4ECA-BEE7-A9F70D3D0E33](https://user-images.githubusercontent.com/71132481/149193898-51470e12-a76c-43ba-be8c-de5abee8f699.png)

**Comment by Ein-Tim** [comment]
@posalex 

This test was never registered before?

**Comment by posalex** [comment]
@Ein-Tim sorry, by bad, the error message says, I can restore the test from the trash bin, so I did, now I can see my tests.

**Comment by posalex** [comment]
![DF298DCF-9D4C-485A-833E-0CBF57B342F6](https://user-images.githubusercontent.com/71132481/149194399-73d13fcc-69ad-4afd-91e4-449dfa4a5985.png)

**Comment by Ein-Tim** [comment]
@posalex 

So everything back to normal again, or?

**Comment by posalex** [comment]
@Ein-Tim in my case, yes. 
I could un-delete the first from the 10th of January test and it automatically updated to the second test from the 12th of January.
As I suggested, they do not generate a new certificate for each test, but do update the first one.
But if somebody deletes the test and deletes it forever, may be they cannot add any tests from this particular testing facility anymore.

**Comment by posalex** [comment]
@Ein-Tim in my case, yes.
I could un-delete the first test from the 10th of January and it automatically updated to the second test from the 12th of January.
As I suggested, they do not generate a new certificate for each test, but do update the first one.
But if somebody deletes the test and deletes it forever, may be they cannot add any tests from this particular testing facility anymore.

**Comment by Ein-Tim** [comment]
@posalex 

I've done this multiple times & had no problems so should you ever encounter this again, please open a new issue! 

Thank you!
